File: DS/binary_search.cpp
/*
    Author: Tajwar Saiyeed
    Date: 2024-11-23 19:26:53
    File: binary_search.cpp
*/

#include <bits/stdc++.h>
using namespace std;

int main()
{
    int n;
    cin >> n;
    int arr[n];
    for (int i = 0; i < n; ++i)
    {
        cin >> arr[i];
    }

    int target;
    cin >> target;

    int low = 0, high = n - 1;
    while (low <= high)
    {
        int mid = low + (high - low) / 2;
        if (arr[mid] == target)
        {
            cout << "Target Found" << endl;
            return 0;
        }
        else if (arr[mid] < target)
        {
            low = mid + 1;
        }
        else
        {
            high = mid - 1;
        }
    }

    cout << "Target Not Found" << endl;
    return 0;
}

File: DS/binary_search.py
numbers = list(map(int, input().split()))
key = int(input())
low = 0
high = len(numbers) - 1

while low <= high:
    mid = (low + high) // 2
    if numbers[mid] == key:
        print("Found", mid)
        break
    elif numbers[mid] < key:
        low = mid + 1
    else:
        high = mid - 1
        
else:
    print("Not Found")
    